From b60e83f0a739795f569f8b9b8d292c63fcec19aa Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Sat, 30 Apr 2011 22:31:39 +0300
Subject: [PATCH] texmath: fix test failure on UTF-8 incompatible locale

Tests assume we consume UTF-8 input. Set it explicitely for base-4.
    $ LANG=C ./setup test
    texmath: <stdin>: hGetContents: invalid argument (Invalid or incomplete multibyte or wide character)
    Test unicode FAILED (< expected, > actual): ...
Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
---
 texmath.hs |   11 ++++++++++-
 1 files changed, 10 insertions(+), 1 deletions(-)

diff --git a/texmath.hs b/texmath.hs
index 764c344..f430499 100644
--- a/texmath.hs
+++ b/texmath.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 module Main where
 
 import Text.TeXMath
@@ -15,9 +16,17 @@ inHtml x =
         unode "meta" ()
     , unode "body" x ]
 
+getUTF8Contents :: IO String
+getUTF8Contents =
+#if MIN_VERSION_base(4,2,0)
+       hSetEncoding stdin utf8 >> hGetContents stdin
+#else
+       hGetContents stdin
+#endif
+
 main :: IO ()
 main = do
-  inp <- getContents
+  inp <- getUTF8Contents
   let (ms, rest) = parseMacroDefinitions inp
   case (texMathToMathML DisplayBlock $! applyMacros ms rest) of
         Left err         -> hPutStrLn stderr err
-- 
1.7.3.4

